xbmc/network/Socket.cpp

```diff
@@ -38,13 +38,38 @@ using namespace SOCKETS;
 /* CPosixUDPSocket                                                    */
 /**********************************************************************/
 
-bool CPosixUDPSocket::Bind(CAddress& addr, int port, int range)
+bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
 {
   // close any existing sockets
   Close();
 
-  // create the socket
-  m_iSock = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+  // If we can, create a socket that works with IPv6 and IPv4.
+  // If not, try an IPv4-only socket (we don't want to end up
+  // with an IPv6-only socket).
+  if (!localOnly) // Only bind loopback to ipv4 to address linux limitations. TODO : Implement dual bindinds.
+  {
+    m_iSock = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+    if (m_iSock != INVALID_SOCKET)
+    {
+#ifdef WINSOCK_VERSION
+      const char zero = 0;
+#else
+      int zero = 0;
+#endif
+      if (setsockopt(m_iSock, IPPROTO_IPV6, IPV6_V6ONLY, &zero, sizeof(&zero)) == -1)
```

Comment:
this does not work for detecting if ipv6 is available or not. I already fixed this somewhere else some years ago:
https://github.com/xbmc/xbmc/pull/3398
you have to bind the socket for reliable detection